### PR TITLE
Set symmetric colorbar limits on HMI maps

### PR DESCRIPTION
**Pull request by dstansby**
We already do this on MDI maps, so it seems sensible to do it for HMI maps too.

xref https://github.com/sunpy/sunpy/issues/5637

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -549,6 +549,14 @@ def __rtruediv__(self, value):
     def _meta_hash(self):
         return self.meta.item_hash()
 
+    def _set_symmetric_vmin_vmax(self):
+        # Set symmetric vmin/vmax about zero
+        vmin = np.nanmin(self.data)
+        vmax = np.nanmax(self.data)
+        threshold = max([abs(vmin), abs(vmax)])
```
Comment (ConorMacBride) [inline sunpy/map/mapbase.py:556]:
```suggestion
        threshold = np.nanmax(abs(self.data))
```

File: changelog/5825.feature.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The colorbar limits on MDI magnetic field maps are now automatically
```
Comment (ConorMacBride) [inline changelog/5825.feature.rst:1]:
```suggestion
The colorbar limits on HMI magnetic field maps are now automatically
```

File: changelog/5825.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Symmetric colorbar limits are no longer set on intensity images from HMI.
```
Comment (ConorMacBride) [inline changelog/5825.bugfix.rst:1]:
```suggestion
Symmetric colorbar limits are no longer set on intensity images from MDI.
```

**Review by ConorMacBride — COMMENTED** [review]
This looks good. Just HMI and MDI are the opposite way around? 

Do we need a figure test for HMI magnetic field maps? The change to HMI hasn't broken any tests. (Probably not, as there doesn't seem to be many figure tests for individual sources.)

File: sunpy/map/sources/soho.py [hunk 4]
```diff
@@ -197,6 +194,13 @@ def _date_obs(self):
         else:
             return parse_time(self.meta['date_obs'])
 
+    @property
+    def unit(self):
+        bunit = self.meta.get('bunit', None)
+        if bunit == 'Arbitrary intensity units':
```
Comment (wtbarnes) [inline sunpy/map/sources/soho.py:200]:
```suggestion
        if bunit is not None and bunit.lower() == 'arbitrary intensity units':
```
Just to be safe

**Review by wtbarnes — APPROVED** [review]

**Comment by nabobalis** [comment]
One of the examples failed with:

```
/home/runner/work/sunpy/sunpy/examples/plotting/magnetogram_active_regions.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/runner/work/sunpy/sunpy/examples/plotting/magnetogram_active_regions.py", line 51, in <module>
    smap.plot(axes=ax, vmin=-120, vmax=120)
  File "/home/runner/work/sunpy/sunpy/.tox/build_docs/lib/python3.9/site-packages/astropy/units/decorators.py", line 304, in wrapper
    return_ = wrapped_function(*func_args, **func_kwargs)
  File "/home/runner/work/sunpy/sunpy/.tox/build_docs/lib/python3.9/site-packages/sunpy/map/mapbase.py", line 2481, in plot
    raise ValueError(msg.format('vmin'))
ValueError: Cannot manually specify vmin, as the norm already has vmin set. To prevent this error set vmin on `m.plot_settings["norm"]` or the norm passed to `m.plot`.
```

**Comment by nabobalis** [comment]
I changed the gallery example, its not pretty but it works locally.

**Review by nabobalis — APPROVED** [review]

File: examples/plotting/magnetogram_active_regions.py [hunk 5]
```diff
@@ -48,7 +48,12 @@
 
 fig = plt.figure()
 ax = fig.add_subplot(projection=smap)
-smap.plot(axes=ax, vmin=-120, vmax=120)
+# Passing vmin/vmax to ``plot`` does not work since
+# a normalisation is set on the map. So we have to
+# work around it like so:
+smap.plot_settings["norm"].vmin=-150
+smap.plot_settings["norm"].vmax=150
```
Comment (wtbarnes) [inline examples/plotting/magnetogram_active_regions.py:55]: What happens if you leave it alone and just let the map source set the symmetric vmin/vmax?

Comment (nabobalis) [reply examples/plotting/magnetogram_active_regions.py:55]: I did that, it sets it to be +-2k and looks very washed out

**Pull request merged**